### ISSUE TITLE
How to know if the TTS training is going good.

### ISSUE THREAD

**Issue by 123srikanth** [espnet/espnet#3874]
Hi there I'm training for espnet/egs/ljspeech/tts1 receipe for hindi tts model. My training stage is started and I just want to know that my training is going good without any flaws that will bring any unappropriate output at the end.  I'm raising this query because earlier also I have trained the model but at the end I didn't get expected output while it was totally vague. So can someone confirm me how to check if my training is going good and at the end I'll get the output atleast to some extent of what I expected. And also there are some graphs generating and how to understand them what they mean.

**Comment by kan-bayashi** [comment]
In the case of TTS, you should carefully check the attention weight plots, which will be found in `exp/your_model/att_ws` directory. If the attention is not diagonal, the model is not trained well.

**Comment by 123srikanth** [comment]
these are the plots that are being generated in training. Some of them are very diagonal while some of them are not and sometimes for the same recording it varied and produced different plots like some are very diagonal and some are not why like these

**Comment by kan-bayashi** [comment]
It seems the diagonal attention is learnt somehow.
But it seems the your audio includes unnecessary silence at the end of audio.
Again, please carefully check these points.
https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/tts1#why-the-model-cannot-be-trained-well-with-my-own-dataset
Careful trimming the silence makes the attention learning more stable.
Or you can reduce the learning rate to make it stable.

**Comment by 123srikanth** [comment]
where can I exactly change the learning rate in the receipe? when I searched for learninrg rate I have got a list of files with learning rate option

/espnet/egs/ljspeech/tts1$ grep -r -l learning_rate *
steps/libs/nnet3/xconfig/attention.py
steps/libs/nnet3/train/chain_objf/acoustic_model.py
steps/libs/nnet3/train/common.py
steps/libs/nnet3/train/frame_level_objf/common.py
steps/info/nnet3_disc_dir_info.pl
steps/train_mmi_fmmi.sh
steps/nnet3/train_discriminative.sh
steps/nnet3/lstm/train.sh
steps/nnet3/train_dnn.py
steps/nnet3/chain/train.py
steps/nnet3/chain/train_tdnn.sh
steps/nnet3/chain/e2e/train_e2e.py
steps/nnet3/tdnn/train.sh
steps/nnet3/tdnn/train_raw_nnet.sh
steps/nnet3/train_tdnn.sh
steps/nnet3/train_rnn.py
steps/nnet3/chain2/internal/get_train_schedule.py
steps/nnet3/train_raw_dnn.py
steps/nnet3/train_raw_rnn.py
steps/train_mmi_fmmi_indirect.sh
steps/nnet2/train_discriminative.sh
steps/nnet2/train_more2.sh
steps/nnet2/train_convnet_accel2.sh
steps/nnet2/train_pnorm_multisplice2.sh
steps/nnet2/train_pnorm.sh
steps/nnet2/train_discriminative_multilang2.sh
steps/nnet2/train_pnorm_accel2.sh
steps/nnet2/train_multilang2.sh
steps/nnet2/retrain_simple2.sh
steps/nnet2/retrain_tanh.sh
steps/nnet2/train_more.sh
steps/nnet2/convert_lda_to_raw.sh
steps/nnet2/train_pnorm_simple2.sh
steps/nnet2/train_pnorm_simple.sh
steps/nnet2/train_tanh_fast.sh
steps/nnet2/train_discriminative2.sh
steps/nnet2/train_pnorm_fast.sh
steps/nnet2/train_tanh_bottleneck.sh
steps/nnet2/train_block.sh
steps/nnet2/update_nnet.sh
steps/nnet2/train_pnorm_bottleneck_fast.sh
steps/nnet2/train_tanh.sh
steps/nnet2/make_multisplice_configs.py
steps/nnet2/train_pnorm_multisplice.sh
steps/nnet2/train_multisplice_ensemble.sh
steps/nnet2/train_pnorm_ensemble.sh
steps/nnet2/train_multisplice_accel2.sh
steps/nnet2/retrain_fast.sh
steps/tfrnnlm/lstm.py
steps/tfrnnlm/vanilla_rnnlm.py
steps/tfrnnlm/lstm_fast.py
utils/nnet-cpu/make_nnet_config.pl
utils/nnet-cpu/make_nnet_config_preconditioned.pl
utils/nnet-cpu/update_learning_rates.pl

**Comment by kan-bayashi** [comment]
It depends on the dataset but it should start from 1/2, 1/4, ....

**Comment by 123srikanth** [comment]
if I use 1/2 it will be more than 1e-3 then it get increased but you have asked to reduce it.

**Comment by kan-bayashi** [comment]
I means the default value * 1/2.

**Comment by 123srikanth** [comment]
okay thank you I'll try that.

**Comment by kan-bayashi** [comment]
@123srikanth Did you clean your data? 
In my experiences, all I need to train good TTS models is cleaning.

**Comment by 123srikanth** [comment]
yes sir I have cleaned it but when I ran the model it is not proceeding further after first stage and this is what I found in logs 

$ cat make_fbank_phn_train.32.log 
# compute-fbank-feats.py --fs 22050 --fmax 7600 --fmin 80 --n_fft 1024 --n_shift 256 --win_length --window hann --n_mels 80 --write-num-frames=ark,t:exp/phn_make_fbank/train/utt2num_frames.32 --compress=true --filetype mat --normalize 16 scp:exp/phn_make_fbank/train/wav.32.scp ark,scp:/home/ubuntu/espnet/egs/ljspeech/tts1/fbank/raw_fbank_phn_train.32.ark,/home/ubuntu/espnet/egs/ljspeech/tts1/fbank/raw_fbank_phn_train.32.scp 
# Started at Sat Dec 18 19:10:33 UTC 2021
#Accounting: time=8 threads=1
# Ended (code 0) at Sat Dec 18 19:10:41 UTC 2021, elapsed time 8 seconds



there is no error I see to sort it out so I went for second option of changing learning rate

**Comment by 123srikanth** [comment]
sir I have trimmed the silence using the script mentioned here https://github.com/espnet/espnet/blob/f03101557753517ebac8c432f0793d97d68fa5f0/egs2/hui_acg/tts1/local/data.sh#L73-L82 but it did not make any changes to my audio files and neither created any new audio files but just created some figures and tmp directories with scp files in each and there is a segment file. How can I use them in my receipe to provide audios without silence.

**Comment by kan-bayashi** [comment]
> just created some figures

You can the figure to check the trimming is performed well. 

> there is a segment file

If the `segments` file is included in data directory, the trimmed audio will be used in extracnting mel-filterbank automatically.
https://github.com/espnet/espnet/blob/682dad150bb13c7d852600583c53ce8d24fd1d6e/egs/ljspeech/tts1/run.sh#L99-L109
https://github.com/espnet/espnet/blob/682dad150bb13c7d852600583c53ce8d24fd1d6e/utils/make_fbank.sh#L94
So you need to re-run the stage to extract fbank.
You can check the log of make_fbank.sh to confirm the use of `segments` file.

**Comment by 123srikanth** [comment]
I have implemented the segments file in my train data and ran the training again and this is what I could see the plots generated. It looks like there is no much difference before and after removing the silence
[plots.zip](https://github.com/espnet/espnet/files/7763124/plots.zip)

**Comment by kan-bayashi** [comment]
> are these okay?

The trimming depends on the threshold parameter. You need to tune by yourself.
But your training curve seems no problem. Let you check the results at first.
